crates/miden-objects/src/account/component/template/storage/placeholder.rs

```diff
@@ -245,48 +296,43 @@ impl TemplateFelt for Felt {
         } else {
             input.parse::<u64>()
         }
-        .map_err(|_| {
-            TemplateTypeError::ParseError(input.to_string(), Self::type_name().to_string())
-        })?;
+        .map_err(|_| TemplateTypeError::ParseError(input.to_string(), Self::type_name()))?;
         Felt::try_from(n).map_err(|_| TemplateTypeError::ConversionError(input.to_string()))
     }
 }
 
 impl TemplateFelt for TokenSymbol {
-    fn type_name() -> &'static str {
-        "tokensymbol"
+    fn type_name() -> TemplateType {
+        TemplateType::new("tokensymbol")
     }
     fn parse_felt(input: &str) -> Result<Felt, TemplateTypeError> {
-        let token = TokenSymbol::new(input).map_err(|_| {
-            TemplateTypeError::ParseError(input.to_string(), Self::type_name().to_string())
-        })?;
+        let token = TokenSymbol::new(input)
+            .map_err(|_| TemplateTypeError::ParseError(input.to_string(), Self::type_name()))?;
         Ok(Felt::from(token))
     }
 }
 
-// WORD IMPLS
+// WORD IMPLS FOR NATIVE TYPES
 // ================================================================================================
 
 impl TemplateWord for Word {
-    fn type_name() -> &'static str {
-        "word"
+    fn type_name() -> TemplateType {
+        TemplateType::new("word")
```

Comment:
Should we use `default_word_type()` here?

Reply:
Done